unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Scripts/WidgetsModule/SettingsWidgetController.cs

```diff
@@ -39,6 +39,7 @@ public class SettingsWidgetController : ScriptableObject, ISettingsWidgetControl
         {
             newControl.Initialize(controlConfig, newControlController);
             controls.Add(newControl);
+            SettingsPanelDataStore.i.controls.controlControllers.Add(newControlController);
```

Comment:
As I left indicated in the PR "annotations", finally I had not to use the `SettingsPanelDataStore` class, so we could remove it if we want.

`SettingsPanelDataStore.i.controls.controlControllers` would be only a collection of controllers that are initially added when the own control is added in the HUD. The initial idea of this was to have some kind of accessor to all our settings controls (only to its controllers) in order to be able to modify them from code without having to go to the HUD panel to physically click on the control. I thought it could be very useful for cases where we need to apply any setting from code and, apart of apply this setting internally, to see visually applied the change in the HUD.

For example: Imagine that in some part of our code we need to change the bloom control setting. We could do this:
`SettingsPanelDataStore.i.TryGetController(out BloomControlController bloomController);`
`bloomController.UpdateSetting(<new value>);`

In this way, the bloom would be changed and we would also see the change reflected in the corresponding settings HUD toogle.

What would be your concern about this kind of solution?

Reply:
I dont like this approach neither, we are trying to move to a Data Driven architecture and minimize the use of singletons.

Why cannot we move the settings model to a new assembly to be referenced by the `DataStore` and use it? We can even have a subclass in the `DataStore` with each setting on its own.

If you wanna change a setting, you go to the DataStore, change it and the systems that need to know about that variable will subscribe to its `OnChange` and react properly.

Reply:
@AjimenezDCL but this is what I exactly doing with this approach, but instead using our general data store, I'm using a specific data store for the settings panel.

Reply:
Anyway guys, if you don't mind, I would prefer to move this discussion to a Post-Weekly and avoid to block this PR. It is something that was implemented in an initial point of this issue but that I have not used later. I simply left the class there for future uses.

Reply:
Bear in mind that with the new environment assembly, `DataStore` will stop being a singleton and will become a reference, so you would have to include this `SettingsPanelDataStore` in `Environment` somehow. 

This approach also have the downside that if you want to change a setting you have to reference the whole `SettingsHUD` assembly, which seems strange.

Moving on and continuing the discussion in a post-daily/weekly it's fine by me.

Reply:
Agree with @AjimenezDCL and move this discussion for later. Ty for the explanation!